caliber/binary_classification/binning/histogram_binning/ppi.py

```diff
@@ -8,34 +8,49 @@
 class PPIHistogramBinningBinaryClassificationModel(
     HistogramBinningBinaryClassificationModel
 ):
-    def fit(self, probs: np.ndarray, targets: np.ndarray, pseudo_indices: np.ndarray):
+    def fit(
+        self,
+        probs: np.ndarray,
+        targets: np.ndarray,
+        pseudo_targets: np.ndarray,
+        labeled_indices: np.ndarray,
+    ):
         self._bin_edges = self._get_bin_edges()
         bin_indices = np.digitize(probs, self._bin_edges)
         self._params = []
 
-        is_pseudo = np.zeros(len(targets), dtype=bool)
-        is_pseudo[pseudo_indices] = True
+        is_labeled = np.zeros(len(targets), dtype=bool)
+        is_labeled[labeled_indices] = True
 
         for i in range(1, self.n_bins + 2):
             mask = bin_indices == i
-            self._fit_bin(i, mask, probs, targets, is_pseudo)
+            self._fit_bin(i, mask, probs, targets, pseudo_targets, is_labeled)
 
     def _fit_bin(
         self,
         i: int,
         mask: np.ndarray,
         probs: np.ndarray,
         targets: np.ndarray,
-        is_pseudo: np.ndarray,
+        pseudo_targets: np.ndarray,
+        is_labeled: np.ndarray,
     ):
         prob_bin = np.mean(mask)
         masked_targets = targets[mask]
-        masked_is_pseudo = is_pseudo[mask]
+        masked_probs = probs[mask]
+        masked_pseudo_targets = pseudo_targets[mask]
+        masked_is_labeled = is_labeled[mask]
 
         self._params.append(
-            np.mean(masked_targets[masked_is_pseudo])
-            + np.mean(
-                masked_targets[~masked_is_pseudo] - masked_targets[~masked_is_pseudo]
+            (
```

Comment:
I suggest using optimal tuning in the estimation process. Here is the code you can use for that (look at example 6.1 in [this paper](https://arxiv.org/pdf/2311.01453). 

```
        n = len(masked_is_labeled)
        N = len(~masked_is_labeled)
        
        if N > 0:
            lambda_estimate = np.cov(masked_targets[masked_is_labeled], masked_pseudo_targets[masked_is_labeled])[0][1] / ((1 + n/N) * np.var(masked_pseudo_targets)) 
            bias_estimate = lambda_estimate * np.mean(masked_pseudo_targets[~masked_is_labeled])
                + np.mean(
                    masked_targets[masked_is_labeled]
                    - lambda_estimate * masked_pseudo_targets[masked_is_labeled]
                )
                - np.mean(masked_probs)
        else:
            bias_estimate = np.mean(masked_targets[masked_is_labeled]) - np.mean(masked_probs)

        self._params.append(
            bias_estimate
            if prob_bin >= self.min_prob_bin
            else np.nan
        )
```